src/client/containers/ConfigurationUpdateContainer/hooks/useConfigurationUpdate.ts

```diff
@@ -58,52 +60,54 @@ export const useConfigurationUpdate = ({
-                            resource === undefined ||
-                            events === undefined ||
-                            workspaceId === undefined) {
-                    saveEvent.notifyFailure("params are not defined");
-                    return;
-                }
-                try {
-                    await updateConfiguration(
-                        {
-                            configurationId,
-                            zeplin: {
-                                resource: {
-                                    id: resource.id,
-                                    type: resource.type
-                                },
-                                workspaceId,
-                                events
-                            }
-                        });
+        if (isInitialized) {
+            microsoftTeams.getContext(({
+                channelId,
+                channelName,
+                tid: tenantId
+            }) => {
+                microsoftTeams.settings.registerOnSaveHandler(async saveEvent => {
+                    if (tenantId === undefined ||
+                        channelId === undefined ||
+                        channelName === undefined ||
+                        resource === undefined ||
+                        events === undefined ||
+                        workspaceId === undefined) {
+                        saveEvent.notifyFailure("params are not defined");
+                        return;
+                    }
+                    try {
+                        await updateConfiguration(
+                            {
+                                configurationId,
+                                zeplin: {
+                                    resource: {
+                                        id: resource.id,
+                                        type: resource.type
+                                    },
+                                    workspaceId,
+                                    events
+                                }
+                            });
 
-                    microsoftTeams.settings.setSettings({
-                        entityId: configurationId,
-                        configName: resource.name,
-                        contentUrl: decodeURI(`${window.location.origin}${url.getConfigurationUpdateUrl({
-                            id: configurationId,
-                            resourceName: resource.name,
-                            resourceType: resource.type,
-                            channel: "{channelName}",
-                            theme: "{theme}"
-                        })}`)
-                    } as microsoftTeams.settings.Settings);
+                        microsoftTeams.settings.setSettings({
+                            entityId: configurationId,
+                            configName: resource.name,
+                            contentUrl: decodeURI(`${window.location.origin}${url.getHomeUrl({
```

Comment:
Since the home page decide which page will be shown. It's safer to use its url as content url.